nixos/modules/services/web-apps/plausible.nix

```diff
@@ -0,0 +1,270 @@
+
+            # Configuration options from
+            # https://plausible.io/docs/self-hosting-configuration
+            PORT = toString cfg.server.port;
+            DISABLE_REGISTRATION = boolToString cfg.server.disableRegistration;
+
+            ADMIN_USER_NAME = cfg.adminUser.name;
+            ADMIN_USER_EMAIL = cfg.adminUser.email;
+
+            DATABASE_SOCKET_DIR = cfg.database.postgres.socket;
+            DATABASE_NAME = cfg.database.postgres.dbname;
+            CLICKHOUSE_DATABASE_URL = cfg.database.clickhouse.url;
+
+            BASE_URL = cfg.server.baseUrl;
+
+            MAILER_EMAIL = cfg.mail.email;
+            SMTP_HOST_ADDR = cfg.mail.smtp.hostAddr;
+            SMTP_HOST_PORT = toString cfg.mail.smtp.hostPort;
+            SMTP_RETRIES = toString cfg.mail.smtp.retries;
+            SMTP_HOST_SSL_ENABLED = boolToString cfg.mail.smtp.enableSSL;
+            ${if cfg.mail.smtp.user != null then "SMTP_USER_NAME" else null} = cfg.mail.smtp.user;
+          };
+
+          path = [ pkgs.plausible ]
+            ++ optional cfg.database.postgres.setup config.services.postgresql.package;
+
+          serviceConfig = {
+            DynamicUser = true;
+            PrivateTmp = true;
+            WorkingDirectory = "/var/lib/plausible";
+            StateDirectory = "plausible";
+            ExecStartPre = "@${pkgs.writeShellScript "plausible-setup" ''
+              ${envSecrets}
+              ${pkgs.plausible}/createdb.sh
+              ${pkgs.plausible}/migrate.sh
+              ${optionalString cfg.adminUser.activate ''
+                if ! ${pkgs.plausible}/init-admin.sh | grep 'already exists'; then
+                  psql -d plausible <<< "UPDATE users SET email_verified=true;"
+                fi
+              ''}
+            ''} plausible-setup";
+            ExecStart = "@${pkgs.writeShellScript "plausible" ''
+              ${envSecrets}
+              plausible start
+            ''} plausible";
+          };
+        };
+      }
+      (mkIf cfg.database.postgres.setup {
+        # Unfortunately `plausible' requires super-user permissions in postgresql, so this
+        # has to be done imperatively here.
+        plausible-postgres = {
+          after = [ "postgresql.service" ];
+          bindsTo = [ "postgresql.service" ];
+          requiredBy = [ "plausible.service" ];
+          partOf = [ "plausible.service" ];
+          serviceConfig.Type = "oneshot";
+          script = ''
+            if [ ! -e /var/lib/plausible/.db-setup ]; then
+              mkdir -p /var/lib/plausible/
+              PSQL() {
+                /run/wrappers/bin/sudo -Hu postgres ${config.services.postgresql.package}/bin/psql --port=5432 "$@"
+              }
+              PSQL -tAc "CREATE EXTENSION IF NOT EXISTS citext;"
+              PSQL -tAc "CREATE ROLE plausible WITH LOGIN;"
+              PSQL -tAc "CREATE DATABASE plausible WITH OWNER plausible;"
+              PSQL -tAc "ALTER USER plausible WITH SUPERUSER;"
```

Comment:
I'll check what exactly this is for, but plausible refuses to start without that, so no matter what I'll find it will make some more patching necessary.

Reply:
I'm also curious about why super user priviledge is required. Just curious though. What is the error when it's not superuser priviledge?

Reply:
Hmm weird, it's an "insufficient privilege" because of a missing right to create a database. I guess I gotta find out why it still wants to create it :/

Reply:
in the migrations, there is probably a "create extention". You need to be superuser to do those.
(I ran into the same problem with another app using postgis for example)
You can probably get away without needing superuser priviledge if you ensure that the extension is there.

Reply:
patched away the superuser requirement :)